modules/shared/nRF52840/inc/system_part1_loader.c

```diff
@@ -45,7 +35,18 @@ void system_part1_init() {
 void system_part2_post_init() __attribute__((alias("system_part1_post_init")));
 
 void system_part1_post_init() {
-    if (is_user_module_valid()) {
+    // NOTE: it may fetch the NCP and radio stack version, so some ealy
+    // initialization work should have been done before getting here.
+    const bool bootloader_validated = HAL_Core_Validate_Modules(1, NULL);
```

Comment:
This change may affect Argon boot up times and application startup times. This needs to be checked (startup SLO tests should help). We may need to backport https://github.com/particle-iot/device-os/pull/2269 and enable for Argon and do the same for the upcoming 3.1.0-rc.1